### PR TITLE
Hotfix/replace travis by circle ci

### LINKED ISSUE #59: Traevis could not build documentation
---
### Describe the bug

Traevis could not build doc

---
### Reproduction scenario

> **note**: You might want to use the Vagrant environment to create a reproducible scenario triggering the bug.

Try to push code on github and see the fail on traevis build

---
### Expected behaviour

The doc could be created

---
### Additional context

> Add any other informations about the problem here. (e.g: logs, screenshots)
```
SNIMissingWarning: An HTTPS request has been made, but the SNI (Server Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  SNIMissingWarning
/home/travis/build/TheFkinCompany/allspark/.docs_venv/local/lib/python2.7/site-packages/pip/_vendor/urllib3/util/ssl_.py:160: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecurePlatformWarning
/home/travis/build/TheFkinCompany/allspark/.docs_venv/local/lib/python2.7/site-packages/pip/_vendor/urllib3/util/ssl_.py:160: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecurePlatformWarning
```

### PR DESCRIPTION
**Pull request by guiadco**
### Current behaviour

Have some bug on travis at this time.

---
### Expected behaviour

Have my github pages updated

---
### Modifications

> What are the changes involved to match with the expected behaviour ?

-  [ ] Add SSH Key to Github
-  [ ] Add SSH Key to Circle CI
-  [ ] Add the circle.yml -> the config file
-  [ ] Add the folder .circleci with the deploy script for gh-pages with Circle CI
-  [ ] I keep the travis file but it could be deprecated in a future version

---
### Related issues

> What issue are affected by or affecting this pull request.

- resolve #59 
- close #60

---
### Status

- [ ] Implementation
- [ ] Test coverage
- [ ] Documentation

### CODE REVIEW

File: .circleci/deploy-ghpages.sh [hunk 1]
```diff
@@ -4,6 +4,7 @@
 # https://gist.github.com/domenic/ec8b0fc8ab45f39403dd
 
 set -xe # Exit with nonzero exit code if anything fails
+env
```
Comment (Xide) [inline .circleci/deploy-ghpages.sh]: Debug command, has nothing to do with the deployment.

File: .circleci/deploy-ghpages.sh [hunk 2]
```diff
@@ -15,7 +16,7 @@ function doCompile {
 }
 
 # Pull requests and commits to other branches shouldn't try to deploy, just build to verify
-if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_BRANCH" != "$SOURCE_BRANCH" ]; then
+if [ "$CIRCLE_PULL_REQUEST" -o "$CIRCLE_BRANCH" != "$SOURCE_BRANCH" ]; then
```
Comment (Xide) [inline .circleci/deploy-ghpages.sh]: If you expect an empty input, you can use `-z`

File: .circleci/deploy-ghpages.sh [hunk 3]
```diff
@@ -42,7 +44,7 @@ doCompile
 
 # Now let's go have some fun with the cloned repo
 cd out
-git config user.name "Travis CI"
+git config user.name "Circle CI"
 git config user.email "$COMMIT_AUTHOR_EMAIL"
```
Comment (Xide) [inline .circleci/deploy-ghpages.sh]: I can't see this variable being set on Circle CI, where did you find it ?

File: .circleci/deploy-ghpages.sh [hunk 4]
```diff
@@ -53,18 +55,7 @@ fi
 
 # Commit the "changes", i.e. the new version.
 # The delta will show diffs between new and old versions.
+#git diff --name-only --diff-filter=M | xargs git add
```
Comment (Xide) [inline .circleci/deploy-ghpages.sh]: Debug comment, not related to the pull request.

File: circle.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    docker:
+      - image: circleci/python:3.6.3
+    environment:
+      - COMMIT_AUTHOR_EMAIL: "guiadco4geekhomeinside@gmail.com"
+    steps:
+      - checkout
+      - run:
+          name: Install Virtualenv dependencies for Pip
+          command: pip install --user virtualenv
+      - run:
```
Comment (Xide) [inline circle.yml]: Why do you put the deployment file in the parent directory ?

File: .circleci/deploy-ghpages.sh (hunk 3 above)
Comment (guiadco) [reply .circleci/deploy-ghpages.sh]: you can found it on circle.yml

**Pull request merged**